scripts/ci/test_extensions.sh

```diff
@@ -34,7 +34,7 @@ for ext in $output; do
     fi
 
     echo "Verifying extension:" $ext
-    az extension add -n $ext
+    az extension add -n $ext --allow-preview True
```

Comment:
How many extensions do not follow the latest version number convention?

The test script should also use the default config. We can make an exception list for outdated extensions.

Reply:
The latest version schema and cli core code is compatible with the version number of all extension modules. 

There are 94 extension modules that only has preview releases (which would be failed when customers use new cli to install extension, and if it fails, the message like following will be prompted to users.) 

![image](https://github.com/Azure/azure-cli/assets/8137253/1c1867cc-5f89-4c7c-9055-4c911a71f799)